File: 125-valid-palindrome/valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {

        erase_if(s, [](char const &c) {
            return !std::isalnum(c);
        });
        string str = "";

        for (int i=0; i< s.length(); i++){
            str += tolower(s[i]);
        }
        
        int sz = str.length()-1;
        for (int i=0; i<str.length()/2; i++){
            if (str[i] != str[sz]){
                return false;
            }
            sz--;
        }
        return true;
    }
};

File: 125-valid-palindrome/valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        if len(s) == 1:
            return True
        s = "".join(char for char in s if char.isalpha() or char.isnumeric()).lower()
        if len(s) % 2 == 0:
            s1 = s[:len(s)//2]
            s2 = "".join(reversed(s[len(s)//2:]))
        else: 
            s1 = s[:len(s)//2]
            s2 = "".join(reversed(s[len(s)//2 + 1:]))
        if s1 == s2:
            return True
        return False
        
        